tests/lib/__init__.py

```diff
@@ -823,54 +821,68 @@ def _git_commit(
 
 
 def _vcs_add(script, version_pkg_path, vcs="git"):
-    if vcs == "git":
-        script.run("git", "init", cwd=version_pkg_path)
-        script.run("git", "add", ".", cwd=version_pkg_path)
-        _git_commit(script, version_pkg_path, message="initial version")
-    elif vcs == "hg":
-        script.run("hg", "init", cwd=version_pkg_path)
-        script.run("hg", "add", ".", cwd=version_pkg_path)
-        script.run(
-            "hg",
-            "commit",
-            "-q",
-            "--user",
-            "pip <distutils-sig@python.org>",
-            "-m",
-            "initial version",
-            cwd=version_pkg_path,
-        )
-    elif vcs == "svn":
-        repo_url = _create_svn_repo(script, version_pkg_path)
-        script.run(
-            "svn", "checkout", repo_url, "pip-test-package", cwd=script.scratch_path
-        )
-        checkout_path = script.scratch_path / "pip-test-package"
-
-        # svn internally stores windows drives as uppercase; we'll match that.
-        checkout_path = checkout_path.replace("c:", "C:")
-
-        version_pkg_path = checkout_path
-    elif vcs == "bazaar":
-        script.run("bzr", "init", cwd=version_pkg_path)
-        script.run("bzr", "add", ".", cwd=version_pkg_path)
-        script.run(
-            "bzr", "whoami", "pip <distutils-sig@python.org>", cwd=version_pkg_path
-        )
-        script.run(
-            "bzr",
-            "commit",
-            "-q",
-            "--author",
-            "pip <distutils-sig@python.org>",
-            "-m",
-            "initial version",
-            cwd=version_pkg_path,
-        )
+    all_vcs: Dict[str, Callable] = {"git": _git,
```

Comment:
As a general rule, a plain `Callable` should not be used. You should, instead, specify the details of what that callable takes as arguments and what it returns.

https://mypy.readthedocs.io/en/stable/kinds_of_types.html#callable-types-and-lambdas